### PR TITLE
Add asset failure debugging

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
[merge]

File: hack/test-assets.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ pushd "${OS_ROOT}" > /dev/null
   echo "Validating checked in bindata.go is up to date..."
   if ! assetdiff=$(diff -u _output/test/assets/bindata.go pkg/assets/bindata.go) ; then
 
-    echo "$assetdiff" | head -c 1000
+    echo "$assetdiff" | head -n 10
```
Comment (mnagy) [inline hack/test-assets.sh]: Maybe more than 10, and add for example `cut -c -250` to avoid enormously long lines.

Comment (liggitt) [inline hack/test-assets.sh]: Binary assets like images tend to be the problematic ones, and get put on a single line, so the first 10 lines gets me both lines from the diff where binary content of one file differs

**Comment by liggitt** [comment]
```
--- FAIL: TestPolicyBasedRestrictionOfBuildConfigStrategies-2 (8.57s)
	build_admission_test.go:227: expected forbidden, got <nil>
```
re[merge]

**Pull request merged**